File: gfg/2024_6_june/c++/12_count_numbers_containing_4.cpp
// TIME COMPLEXITY
// O(nlogn)

// SPACE COMPLEXITY
// O(n)

// CODE

class Solution {
  public:
    int countNumberswith4(int n) {
        int cnt = 0;
        for(int i = 1; i <= n; i++){
            string s = to_string(i);
            for(char ch : s) {
                if(ch == '4') {
                    cnt++;
                    break;
                }
            }
        }
        return cnt;
    }
};

File: gfg/2024_6_june/python/12_count_numbers_containing_4.py
# CODE

class Solution:
    def countNumberswith4(self, n : int) -> int:
        if n < 4:
            return 0
        cnt = 0
        for i in range(4, n + 1):
            s = str(i)
            if "4" in s:
                cnt += 1
        return cnt